config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto

```diff
@@ -88,11 +88,29 @@ message GetAllConfigsRequest {
 
   // required - namespace with which the config resource is associated
   string resource_namespace = 2;
+
+  // optional - filtering criteria to narrow down the configs.
+  // Supports relational and logical operators on config fields.
+  optional Filter filter = 3;
+
+  // optional - list of sorting conditions to order the results.
+  // Multiple SortBy entries are applied in the specified order of priority.
+  repeated SortBy sort_by = 4;
+
+  // optional - pagination parameters to limit and offset the result set.
+  // Useful for retrieving configs in pages when total count is large.
+  optional Pagination pagination = 5;
 }
 
 message GetAllConfigsResponse {
   // list of config values along with the associated contexts, sorted in the descending order of their creation time
   repeated ContextSpecificConfig context_specific_configs = 1;
+
+  // Total number of records matching the filter before pagination
+  int32 total_count = 2;
```

Comment:
This can be expensive to compute and requires a second query. If we need it, make it optional with its presence explicitly requested by the caller.

Reply:
Added boolean in request

Reply:
```suggestion
  optional int32 total_count = 2;
```

Since this can be unset and is a primitive, we want to distinguish between not setting it and it just being 0 matches